### PR TITLE
Spleeter 2.1.0

### PR DESCRIPTION
**Pull request by Faylixe**
Mostly code cleaning and transition to modern Python, including:

- [x] Python type annotations
- [x] Docstring refactor
- [X] FFMpeg time formatting fix
- [X] Switch from `requests` to `httpx` with _HTTP/2_ support for model downloading
- [x] Switch from `argparse` to `typer`
- [x] Various code design clean up

### CODE REVIEW

**Comment by Faylixe** [comment]
API changelog:

- function `get_default_audio_adapter` is now available as `default()` class method within `AudioAdapter` class
- function `get_default_model_provider` is now available as `default()` class method within `ModelProvider` class
- `STFTBackend` and `Codec` are now string enum
- `GithubModelProvider` now use `httpx` with HTTP/2 support
- Commands are now located in `__main__` module, wrapped as simple function using Typer

File: spleeter/audio/__init__.py [hunk 1]
```diff
@@ -10,6 +10,42 @@
...
+
+class Codec(str, Enum):
+    """ Enumeration of supported audio codec. """
+
+    WAV: str = 'wav'
+    MP3: str = 'mp3'
+    OGG: str = 'ogg'
+    M4A: str = 'm4a'
+    WMA: str = 'wma'
+    FLAC: str = 'flac'
```
Comment (faroit) [inline spleeter/audio/__init__.py:33]: while you are at it, i would propose to "decouple" the codec from the container and reflect this change in the API

**Review by faroit — COMMENTED** [review]
Hi, i was just seeing this and I thought it that it might be appropiate to add some comments:

I talked with @mmoussallam in #357 to add a new adapter to spleeter based on a new version of [stempeg](https://github.com/faroit/stempeg/pull/28). 

So one thing I am curious about is the changes you made to the ffmpeg time helper function as you now switched to `str(dt.timedelta(seconds=duration))` - I copied this function to stempeg. Is there a bug regarding the use of this function?

Furthermore see my comment about the codec vs container thing... See my comment [here](https://github.com/faroit/stempeg/pull/28#issuecomment-623114880)

File: spleeter/audio/__init__.py (hunk 1 above)
Comment (Faylixe) [reply spleeter/audio/__init__.py:33]: Hey thanks for you comment !

Actually this is an issue we addressed internally on other projects as we are using same Adapter pattern with FFMPEG, and one of our team member noticed some edge case were the previous implementation returned wrong values which lead to FFMPEG returning empty arrays.

We will perform a full review of this PR once it is finished, I would recommend you to wait before to back port this patch to your adapter just to be sure but I think you should indeed do it at the end :).

Concerning the codec vs container issue this is indeed maybe the right place to setup such thing or at least to start thinking about it. This is quite a heavy PR as we are performing lot of changes without breaking compatibility. But switching to Typer actually will make the process of adding or updating command line options a lot more easier so once it is done we should be able to address this problem in a second iteration probably. We will discuss that with the team this week and let you know.

Comment (faroit) [reply spleeter/audio/__init__.py:33]: @Faylixe sounds good.

Maybe related: I cover an additional ffmpeg time issue with very small numbers in my adapter: https://github.com/faroit/stempeg/blob/master/stempeg/read.py#L193-L197

not sure if `str(dt.timedelta(seconds=duration))` makes this unnecessary, but better safe than sorry ;-)

**Comment by Faylixe** [comment]
Everything sounds good, tests are passing and new CLI interface successfully tested for separation with `audio_example`.

Waiting for review before to deploy.

File: spleeter/audio/__init__.py (hunk 1 above)
Comment (Faylixe) [reply spleeter/audio/__init__.py:33]: Nice, well I think the standard python library should handle precision but to be tested to ensure that with the new version.

File: spleeter/options.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+__email__ = 'spleeter@deezer.com'
+__author__ = 'Deezer Research'
+__license__ = 'MIT License'
+
+AudioInputArgument: ArgumentInfo = Argument(
+    ...,
+    help='List of input audio file path',
+    exists=True,
+    file_okay=True,
```
Comment (romi1502) [inline spleeter/options.py:23]: It seems the API has changed here, with removal of the "-i" option for input files.

File: spleeter/audio/__init__.py (hunk 1 above)
Comment (faroit) [reply spleeter/audio/__init__.py:28]: i tested it in spleeter, and things are good!

**Comment by Faylixe** [comment]
Alright everything is good. Just writing here for history:

- We decided to drop the `spleeter-gpu` package as it is not relevant anymore
- We won't maintain an additional `spleeter-cpu` package to match the new Tensorflow release model
- `-i` option is now deprecated in favor of input args list
- Project is now managed using Poetry (CI updated accordingly)
- Finally code formatting is required using Black and iSort

An additional PR will be done later to introduce type checking using MyPy and dedicated stub file for autocompletion with TypeShred.

**Pull request merged**